OnDiskKeySerializer.deserializeKeySize() is wrong

### Description

The implementation of deserializeKeySize() does the wrong thing, as size is written as int in first 4 bytes all is needed is to read that size. This results in exceptions like:

```
java.lang.RuntimeException: java.lang.IllegalArgumentException: newPosition > limit: (134 > 130)
        at com.swirlds.merkledb.files.hashmap.HalfDiskHashMap.endWriting(HalfDiskHashMap.java:493) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.merkledb.MerkleDbDataSource.writeLeavesToPathToKeyValue(MerkleDbDataSource.java:1305) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.merkledb.MerkleDbDataSource.saveRecords(MerkleDbDataSource.java:570) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.virtualmap.internal.merkle.VirtualRootNode.flush(VirtualRootNode.java:1135) ~[swirlds-virtualmap-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.virtualmap.internal.merkle.VirtualRootNode.flush(VirtualRootNode.java:1115) ~[swirlds-virtualmap-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.virtualmap.internal.pipeline.VirtualPipeline.flush(VirtualPipeline.java:497) ~[swirlds-virtualmap-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.virtualmap.internal.pipeline.VirtualPipeline.hashFlushMerge(VirtualPipeline.java:557) ~[swirlds-virtualmap-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.virtualmap.internal.pipeline.VirtualPipeline.doWork(VirtualPipeline.java:575) ~[swirlds-virtualmap-0.42.0-SNAPSHOT.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IllegalArgumentException: newPosition > limit: (134 > 130)
        at java.nio.Buffer.createPositionException(Buffer.java:341) ~[?:?]
        at java.nio.Buffer.position(Buffer.java:316) ~[?:?]
        at java.nio.ByteBuffer.position(ByteBuffer.java:1516) ~[?:?]
        at com.swirlds.merkledb.files.hashmap.Bucket.findEntryOffset(Bucket.java:380) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.merkledb.files.hashmap.Bucket.putValue(Bucket.java:253) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.merkledb.files.hashmap.BucketMutation.forEachKeyValue(BucketMutation.java:85) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.merkledb.files.hashmap.HalfDiskHashMap.readUpdateQueueBucket(HalfDiskHashMap.java:544) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
        at com.swirlds.merkledb.files.hashmap.HalfDiskHashMap.lambda$endWriting$2(HalfDiskHashMap.java:481) ~[swirlds-merkledb-0.42.0-SNAPSHOT.jar:?]
```

### Steps to reproduce

Run with large enough data set to make sure virtual node cache is exceeded and we really write to disk.

### Additional context

_No response_

### Hedera network

other

### Version

0.0.43

### Operating system

None

--- a/hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKeySerializer.java
+++ b/hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKeySerializer.java
@@ -106,7 +106,7 @@ public final class OnDiskKeySerializer<K> implements KeySerializer<OnDiskKey<K>>
 
     @Override
     public int deserializeKeySize(@NonNull final ByteBuffer byteBuffer) {
-        return byteBuffer.getInt();
+        return byteBuffer.getInt() + Integer.BYTES;
     }
 
     @Override

--- a/hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKeySerializer.java
+++ b/hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskKeySerializer.java
@@ -106,12 +106,7 @@ public final class OnDiskKeySerializer<K> implements KeySerializer<OnDiskKey<K>>
 
     @Override
     public int deserializeKeySize(@NonNull final ByteBuffer byteBuffer) {
-        try {
-            return codec.measure(BufferedData.wrap(byteBuffer)) + 4;
-        } catch (IOException e) {
-            // Maybe log here?
-            return -1;
-        }
+        return byteBuffer.getInt();
     }
 
     @Override

